railties/lib/rails/generators/actions.rb

```diff
@@ -221,14 +222,17 @@ def initializer(filename, data = nil)
       # the generator or an Array that is joined.
       #
       #   generate(:authenticated, "user session")
-      def generate(what, *args)
-        log :generate, what
-
+      def generate(*args)
         options = args.extract_options!
-        options[:without_rails_env] = true
-        argument = args.flat_map(&:to_s).join(" ")
+        args = Shellwords.split(args.join(" "))
+
+        log :generate, args.first
 
-        execute_command :rails, "generate #{what} #{argument}", options
+        in_root do
+          silence_warnings do
```

Comment:
Why do we want to silence warnings here?

Reply:
`Rails::Command.invoke` may load Ruby files which produce warnings ([example](https://buildkite.com/rails/rails/builds/64924#ce9bfb59-722f-4b81-90ee-b2fdc279a618/954-956)).  These warnings are naturally swallowed when shelling out, so I silenced them here to mimic that behavior.